src/System.Management.Automation/utils/CryptoUtils.cs

```diff
@@ -627,18 +627,6 @@ protected void Dispose(bool disposing)
             }
         }
 
-        /// <summary>
-        /// Destructor.
-        /// </summary>
-        ~PSRSACryptoServiceProvider()
-        {
-            // When Dispose() is called, GC.SuppressFinalize()
-            // is called and therefore this finalizer will not
-            // be invoked. Hence this is run only on process
-            // shutdown
-            Dispose(true);
```

Comment:
It is not clear why we need to run the finalizer on process shutdown. I guess it comes from .Net Framework specific.

Reply:
I agree that we should not run `Dispose(true)` on the finalizer thread, since object references are not valid at that time.  The crypto objects themselves should implement a finalizer.